src/utils/validate-schema.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Checks if Valid default coordinates
+ * @param {unknown} possibleCoordinates
+ * @param {number} dimensions - 2 for 2D Vector or 3 for 3D vector
+ * @returns {boolean} A boolean determining if coordinates are parsed correctly.
+ */
+function isValidDefaultCoordinate (possibleCoordinates, dimensions) {
+  if (typeof possibleCoordinates !== 'object' || possibleCoordinates === null) {
```

Comment:
Hmm...that does make sense. Since currently if you do not define a default value, when you register a component it will register a default value at `{ x:0, y:0, z:0 }` If you don't want that to happen assigning it to `null` makes sense. I'll adjust it.